sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -607,14 +607,21 @@ class Analyzer(
     def apply(plan: LogicalPlan): LogicalPlan = plan resolveOperators {
       // Skip sort with aggregate. This will be handled in ResolveAggregateFunctions
       case sa @ Sort(_, _, child: Aggregate) => sa
-
+      case sa @ Sort(_, _, _)
+        if sa.order.exists(so => ResolveAggregateFunctions.containsAggregate(so.child)) => sa
       case s @ Sort(order, _, child) if !s.resolved && child.resolved =>
         try {
           val newOrder = order.map(resolveExpressionRecursively(_, child).asInstanceOf[SortOrder])
           val requiredAttrs = AttributeSet(newOrder).filter(_.resolved)
           val missingAttrs = requiredAttrs -- child.outputSet
-          if (missingAttrs.nonEmpty) {
-            // Add missing attributes and then project them away after the sort.
+          // resolve the unresolved functions for knowing if they are aggregate functions
+          val evaluatedSort =
+            ResolveFunctions.resolveFunctions(Sort(newOrder, s.global, child)).asInstanceOf[Sort]
+          if (evaluatedSort.order.exists(so =>
+              ResolveAggregateFunctions.containsAggregate(so.child))) {
```

Comment:
Can we move this part of resolving functions of Sort as another pattern matching case? So this original pattern matching case can be untouched? I think it will more clear.